usl_pipeline/study_area_uploader/study_area_uploader/main.py

```diff
@@ -76,21 +98,38 @@ def main() -> None:
             else:
                 break
 
-        logging.info(
-            "Storage bucket for study area chunks: %s",
-            cloud_storage.STUDY_AREA_CHUNKS_BUCKET,
-        )
         study_area_chunkers.build_and_upload_chunks(
             args.name,
             prepared_inputs,
             work_dir,
-            storage_client.bucket(cloud_storage.STUDY_AREA_CHUNKS_BUCKET),
+            chunk_bucket,
             args.chunk_length,
             input_elevation_band=args.elevation_geotiff_band,
         )
 
 
-def parse_args() -> argparse.Namespace:
+def _check_and_delete_storage_files_with_prefix(
+    bucket: storage.Bucket, prefix: str, override: bool
+) -> None:
+    """Checks if storage bucket folder has files, deletes if allowed or stops if not."""
+    blobs = [blob for blob in bucket.list_blobs(prefix=f"{prefix}")]
+    if len(blobs) == 0:
+        # No files, nothing to check or delete
+        return
+    if not override:
+        # Deletion is not allowed, stop the execution
+        raise ValueError(
```

Comment:
I think it'd be better to print a message and exit cleanly in this case rather than raise an exception. The stack trace will make it look like something broke to the user.

Reply:
You can use [sys.exit](https://docs.python.org/3/library/sys.html#sys.exit) to return non-zero from the script (although I think it's bad form to call sys.exit outside of `main`)

Reply:
Yep, I refactored that function to print error to logs and return False. In main I do `sys.exit` every time this function returns False.